python/prairielearn.py

```diff
@@ -688,6 +688,17 @@ def get_float_attrib(element, name, *args):
     return float_val
 
 
+# Order here matters, as we want to override the case where the args is omitted
+@overload
+def get_color_attrib(element: lxml.html.HtmlElement, name: str) -> str:
+    ...
+
+
+@overload
+def get_color_attrib(element: lxml.html.HtmlElement, name: str, *args: str) -> str:
+    ...
```

Comment:
I would prefer using `pass` over `...` here for consistency with the rest of the codebase (its the same thing right? I've actually never seen `...` used before).

I'm also not very familiar with the `@overload` construct. Is there a simpler way we can achieve the same thing? I actually don't see where this is being used at all because it seems like the third argument is always being passed in whenever this function is being called.

Reply:
The use of `...` is the preferred syntax when using the `@overload` decorator (see [here](https://docs.python.org/3/library/typing.html#typing.overload)), and the usage here is consistent with other places in this file.

Unfortunately I don't think there's a nice way of simplifying this, type annotations don't play super nice with the way this function has its optional arguments set up. In particular, the return type can change depending on what default argument gets passed in.

Reply:
Can you point me to where this version of the function is actually called? All the calls that I saw explicitly had all three arguments.

Reply:
You know what, I actually don't think the version with `*args` omitted is needed for any of these functions (since `*args` is always treated as optional). I'll remove here and clean that up for the other functions in a follow-up.